tests/util/test_json.py

```diff
@@ -71,9 +74,12 @@ def test_overwrite_and_reload():
 
 def test_save_bad_data():
     """Test error from trying to save unserialisable data."""
-    fname = _path_for("test4")
-    with pytest.raises(SerializationError):
-        save_json(fname, TEST_BAD_OBJECT)
+    with pytest.raises(SerializationError) as excinfo:
+        save_json("test4", {"hello": set()})
+
+    assert "Failed to serialize to JSON: test4. Bad data found at $.hello" in str(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5020228)